src/Microsoft.AspNetCore.Mvc.TagHelpers/PartialTagHelper.cs

```diff
@@ -104,15 +129,36 @@ public object Model
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var viewEngineResult = FindView();
+            if (Optional && _hasFallback)
+            {
+                throw new InvalidOperationException(
+                    Resources.FormatPartialTagHelper_InvalidModelAttributes(
+                    typeof(PartialTagHelper).FullName,
+                    FallbackAttributeName,
+                    OptionalAttributeName));
+            }
 
-            if (viewEngineResult.Success)
+            ViewEngineResult result;
+            if (Optional)
```

Comment:
If we allow `Optional` to be applied with fallback, could this be simplified?

Reply:
Yes, if optional is allowed then code can be simplified a lot.